torch/lib/THNN/generic/LogSigmoid.c

```diff
@@ -10,11 +10,12 @@ void THNN_(LogSigmoid_updateOutput)(
 {
   THTensor_(resizeAs)(output, input);
   THTensor_(resizeAs)(buffer, input);
-
+  //Use the LogSumExp trick to make this stable against overflow
   TH_TENSOR_APPLY3(real, output, real, input, real, buffer,
-    real z = exp(-*input_data);
+    real max_elem = fmax(0, -*input_data);
+    real z = exp(-max_elem) + exp(-*input_data - max_elem);
```

Comment:
nit: For another PR, it would be good to use the appropriate `exp`/`expf` functions depending on `real` type.
Something in the lines of https://github.com/torch/torch7/pull/1028/files for example